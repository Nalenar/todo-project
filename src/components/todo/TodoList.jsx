import { useSelector } from "react-redux";

import TodoAddForm from "./TodoAddForm";
import TodoItem from "./TodoItem";
import EditListForm from "./EditListForm";
import Dropdown from "../common/Dropdown";
import Modal from "../common/Modal";
import EditTodo from "./EditTodo";

const TodoList = (props) => {
  const todos = useSelector((state) => state.todos);
  const blur = useSelector((state) => state.style.styles.blur);
  const editList = useSelector((state) => state.style.windows.editList);

  return (
    <div className={`overflow-y-auto ${blur}`}>
      <div className="flex h-[100px] items-center justify-center">
        {editList ? (
          <EditListForm id={props.category} title={props.headerText} />
        ) : (
          <>
            <h2 className="mr-4 text-center font-roboto text-3xl font-bold text-blue-800">
              {props.headerText}
            </h2>
            {props.list_edit ? <Dropdown /> : <></>}
          </>
        )}
      </div>

      <TodoAddForm category={props.category} />

      <Modal id={props.category} />

      {todos
        .filter((todo) => todo.category === props.category)
        .map((todo) => (
          <TodoItem key={todo.id} todo={todo} />
        ))}
    </div>
  );
};

export default TodoList;
