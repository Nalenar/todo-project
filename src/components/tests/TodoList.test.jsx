import { it, expect, describe, vi } from "vitest";
import { render } from "@testing-library/react";
import * as reduxHooks from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";

import TodoList from "../todo/TodoList";

vi.mock("react-redux");

const mockedSelector = vi.spyOn(reduxHooks, "useSelector");

const todos = [
  {
    id: 1,
    title: "today",
    completed: true,
    list: ["today"],
  },
  {
    id: 2,
    title: "important",
    completed: false,
    list: ["important"],
  },
];

const blur = true;

describe("TodoList", () => {
  it("should create TodoList with empty todos", () => {
    mockedSelector.mockReturnValue([]);

    const component = render(
      <TodoList listId="tasks" listTitle="Задачи" listEdit={false} />,
      { wrapper: Router }
    );

    expect(component).toMatchSnapshot();
  });

  it("should create TodoList with todo items", () => {
    mockedSelector.mockReturnValue(todos);

    const component = render(
      <TodoList listId="tasks" listTitle="Задачи" listEdit={false} />,
      { wrapper: Router }
    );

    expect(component).toMatchSnapshot();
  });

  it("shoud create Todolist with todo items in today list", () => {
    mockedSelector.mockReturnValue(todos);

    const component = render(
      <TodoList listId="today" listTitle="Мой день" listEdit={false} />,
      { wrapper: Router }
    );

    expect(component).toMatchSnapshot();
  });
});
