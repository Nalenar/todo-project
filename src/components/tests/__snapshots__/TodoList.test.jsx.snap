// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoList > shoud create Todolist with todo items in today list 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overflow-y-auto [object Object],[object Object]"
      >
        <div
          class="flex h-[100px] items-center justify-center"
        >
          <form
            class="flex items-center justify-center"
          >
            <input
              class="form-input mr-3 h-10 w-52 border-none bg-transparent p-1 text-center font-roboto text-2xl placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-500"
              placeholder="Мой день"
              type="text"
              value=""
            />
            <button
              class="h-10 rounded-md bg-blue-500 px-2 py-1 font-roboto text-xl text-white shadow-md hover:bg-blue-600"
              type="submit"
            >
              Change
            </button>
          </form>
        </div>
        <form
          class="mx-3 mb-2 overflow-y-hidden rounded border border-gray-200 bg-white shadow-md h-12"
        >
          <div
            class="flex items-center py-1"
          >
            <div
              class="flex h-full w-12 items-center justify-center"
            >
              <span />
            </div>
            <div
              class="w-full"
            >
              <input
                class="form-input w-full border-none font-roboto placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-300"
                name="todo"
                placeholder="Добавьте задачу"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="invisible"
          >
            <hr />
            <div
              class="flex h-10 items-center justify-end bg-gray-100"
            >
              <span />
              <span />
              <span />
              <button
                class="mx-3 rounded-md border bg-white px-2 py-[2px] font-roboto text-blue-500 hover:bg-blue-700 hover:text-white "
                type="submit"
              >
                Добавить
              </button>
            </div>
          </div>
        </form>
        <div
          class="fixed left-1/2 top-1/2 z-[2] -my-36 -ml-40 flex h-36 w-80 flex-col justify-center rounded-md border-2 border-blue-500 bg-white shadow-lg transition-all duration-200 scale-100"
        >
          <h1
            class="mb-3 text-center font-roboto text-lg font-medium"
          >
            Вы действительно хотите удалить этот список?
          </h1>
          <div
            class="flex flex-row items-center justify-around font-bold"
          >
            <a
              class="mr-2 rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto text-red-500 hover:border-blue-500 focus:underline focus:underline-offset-8"
              href="/app/"
            >
              Да
            </a>
            <button
              class="rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto font-bold hover:border hover:border-blue-400 focus:underline focus:underline-offset-8"
            >
              Нет
            </button>
          </div>
        </div>
        <div
          class="mx-3 mt-3 flex h-12 cursor-pointer items-center rounded border border-gray-200 bg-white shadow-md last-of-type:mb-2"
        >
          <div
            class="mx-3 cursor-pointer"
          >
            <span />
          </div>
          <p
            class="flex h-full w-full items-center overflow-x-auto italic text-gray-400 line-through"
          >
            today
          </p>
          <button
            class="mx-3 flex h-[30px] w-[30px] cursor-pointer items-center justify-center"
          >
            <span />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overflow-y-auto [object Object],[object Object]"
    >
      <div
        class="flex h-[100px] items-center justify-center"
      >
        <form
          class="flex items-center justify-center"
        >
          <input
            class="form-input mr-3 h-10 w-52 border-none bg-transparent p-1 text-center font-roboto text-2xl placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-500"
            placeholder="Мой день"
            type="text"
            value=""
          />
          <button
            class="h-10 rounded-md bg-blue-500 px-2 py-1 font-roboto text-xl text-white shadow-md hover:bg-blue-600"
            type="submit"
          >
            Change
          </button>
        </form>
      </div>
      <form
        class="mx-3 mb-2 overflow-y-hidden rounded border border-gray-200 bg-white shadow-md h-12"
      >
        <div
          class="flex items-center py-1"
        >
          <div
            class="flex h-full w-12 items-center justify-center"
          >
            <span />
          </div>
          <div
            class="w-full"
          >
            <input
              class="form-input w-full border-none font-roboto placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-300"
              name="todo"
              placeholder="Добавьте задачу"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="invisible"
        >
          <hr />
          <div
            class="flex h-10 items-center justify-end bg-gray-100"
          >
            <span />
            <span />
            <span />
            <button
              class="mx-3 rounded-md border bg-white px-2 py-[2px] font-roboto text-blue-500 hover:bg-blue-700 hover:text-white "
              type="submit"
            >
              Добавить
            </button>
          </div>
        </div>
      </form>
      <div
        class="fixed left-1/2 top-1/2 z-[2] -my-36 -ml-40 flex h-36 w-80 flex-col justify-center rounded-md border-2 border-blue-500 bg-white shadow-lg transition-all duration-200 scale-100"
      >
        <h1
          class="mb-3 text-center font-roboto text-lg font-medium"
        >
          Вы действительно хотите удалить этот список?
        </h1>
        <div
          class="flex flex-row items-center justify-around font-bold"
        >
          <a
            class="mr-2 rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto text-red-500 hover:border-blue-500 focus:underline focus:underline-offset-8"
            href="/app/"
          >
            Да
          </a>
          <button
            class="rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto font-bold hover:border hover:border-blue-400 focus:underline focus:underline-offset-8"
          >
            Нет
          </button>
        </div>
      </div>
      <div
        class="mx-3 mt-3 flex h-12 cursor-pointer items-center rounded border border-gray-200 bg-white shadow-md last-of-type:mb-2"
      >
        <div
          class="mx-3 cursor-pointer"
        >
          <span />
        </div>
        <p
          class="flex h-full w-full items-center overflow-x-auto italic text-gray-400 line-through"
        >
          today
        </p>
        <button
          class="mx-3 flex h-[30px] w-[30px] cursor-pointer items-center justify-center"
        >
          <span />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TodoList > should create TodoList with empty todos 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overflow-y-auto "
      >
        <div
          class="flex h-[100px] items-center justify-center"
        >
          <form
            class="flex items-center justify-center"
          >
            <input
              class="form-input mr-3 h-10 w-52 border-none bg-transparent p-1 text-center font-roboto text-2xl placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-500"
              placeholder="Задачи"
              type="text"
              value=""
            />
            <button
              class="h-10 rounded-md bg-blue-500 px-2 py-1 font-roboto text-xl text-white shadow-md hover:bg-blue-600"
              type="submit"
            >
              Change
            </button>
          </form>
        </div>
        <form
          class="mx-3 mb-2 overflow-y-hidden rounded border border-gray-200 bg-white shadow-md h-12"
        >
          <div
            class="flex items-center py-1"
          >
            <div
              class="flex h-full w-12 items-center justify-center"
            >
              <span />
            </div>
            <div
              class="w-full"
            >
              <input
                class="form-input w-full border-none font-roboto placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-300"
                name="todo"
                placeholder="Добавьте задачу"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="invisible"
          >
            <hr />
            <div
              class="flex h-10 items-center justify-end bg-gray-100"
            >
              <span />
              <span />
              <span />
              <button
                class="mx-3 rounded-md border bg-white px-2 py-[2px] font-roboto text-blue-500 hover:bg-blue-700 hover:text-white "
                type="submit"
              >
                Добавить
              </button>
            </div>
          </div>
        </form>
        <div
          class="fixed left-1/2 top-1/2 z-[2] -my-36 -ml-40 flex h-36 w-80 flex-col justify-center rounded-md border-2 border-blue-500 bg-white shadow-lg transition-all duration-200 scale-100"
        >
          <h1
            class="mb-3 text-center font-roboto text-lg font-medium"
          >
            Вы действительно хотите удалить этот список?
          </h1>
          <div
            class="flex flex-row items-center justify-around font-bold"
          >
            <a
              class="mr-2 rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto text-red-500 hover:border-blue-500 focus:underline focus:underline-offset-8"
              href="/app/"
            >
              Да
            </a>
            <button
              class="rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto font-bold hover:border hover:border-blue-400 focus:underline focus:underline-offset-8"
            >
              Нет
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overflow-y-auto "
    >
      <div
        class="flex h-[100px] items-center justify-center"
      >
        <form
          class="flex items-center justify-center"
        >
          <input
            class="form-input mr-3 h-10 w-52 border-none bg-transparent p-1 text-center font-roboto text-2xl placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-500"
            placeholder="Задачи"
            type="text"
            value=""
          />
          <button
            class="h-10 rounded-md bg-blue-500 px-2 py-1 font-roboto text-xl text-white shadow-md hover:bg-blue-600"
            type="submit"
          >
            Change
          </button>
        </form>
      </div>
      <form
        class="mx-3 mb-2 overflow-y-hidden rounded border border-gray-200 bg-white shadow-md h-12"
      >
        <div
          class="flex items-center py-1"
        >
          <div
            class="flex h-full w-12 items-center justify-center"
          >
            <span />
          </div>
          <div
            class="w-full"
          >
            <input
              class="form-input w-full border-none font-roboto placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-300"
              name="todo"
              placeholder="Добавьте задачу"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="invisible"
        >
          <hr />
          <div
            class="flex h-10 items-center justify-end bg-gray-100"
          >
            <span />
            <span />
            <span />
            <button
              class="mx-3 rounded-md border bg-white px-2 py-[2px] font-roboto text-blue-500 hover:bg-blue-700 hover:text-white "
              type="submit"
            >
              Добавить
            </button>
          </div>
        </div>
      </form>
      <div
        class="fixed left-1/2 top-1/2 z-[2] -my-36 -ml-40 flex h-36 w-80 flex-col justify-center rounded-md border-2 border-blue-500 bg-white shadow-lg transition-all duration-200 scale-100"
      >
        <h1
          class="mb-3 text-center font-roboto text-lg font-medium"
        >
          Вы действительно хотите удалить этот список?
        </h1>
        <div
          class="flex flex-row items-center justify-around font-bold"
        >
          <a
            class="mr-2 rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto text-red-500 hover:border-blue-500 focus:underline focus:underline-offset-8"
            href="/app/"
          >
            Да
          </a>
          <button
            class="rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto font-bold hover:border hover:border-blue-400 focus:underline focus:underline-offset-8"
          >
            Нет
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TodoList > should create TodoList with todo items 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overflow-y-auto [object Object],[object Object]"
      >
        <div
          class="flex h-[100px] items-center justify-center"
        >
          <form
            class="flex items-center justify-center"
          >
            <input
              class="form-input mr-3 h-10 w-52 border-none bg-transparent p-1 text-center font-roboto text-2xl placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-500"
              placeholder="Задачи"
              type="text"
              value=""
            />
            <button
              class="h-10 rounded-md bg-blue-500 px-2 py-1 font-roboto text-xl text-white shadow-md hover:bg-blue-600"
              type="submit"
            >
              Change
            </button>
          </form>
        </div>
        <form
          class="mx-3 mb-2 overflow-y-hidden rounded border border-gray-200 bg-white shadow-md h-12"
        >
          <div
            class="flex items-center py-1"
          >
            <div
              class="flex h-full w-12 items-center justify-center"
            >
              <span />
            </div>
            <div
              class="w-full"
            >
              <input
                class="form-input w-full border-none font-roboto placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-300"
                name="todo"
                placeholder="Добавьте задачу"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="invisible"
          >
            <hr />
            <div
              class="flex h-10 items-center justify-end bg-gray-100"
            >
              <span />
              <span />
              <span />
              <button
                class="mx-3 rounded-md border bg-white px-2 py-[2px] font-roboto text-blue-500 hover:bg-blue-700 hover:text-white "
                type="submit"
              >
                Добавить
              </button>
            </div>
          </div>
        </form>
        <div
          class="fixed left-1/2 top-1/2 z-[2] -my-36 -ml-40 flex h-36 w-80 flex-col justify-center rounded-md border-2 border-blue-500 bg-white shadow-lg transition-all duration-200 scale-100"
        >
          <h1
            class="mb-3 text-center font-roboto text-lg font-medium"
          >
            Вы действительно хотите удалить этот список?
          </h1>
          <div
            class="flex flex-row items-center justify-around font-bold"
          >
            <a
              class="mr-2 rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto text-red-500 hover:border-blue-500 focus:underline focus:underline-offset-8"
              href="/app/"
            >
              Да
            </a>
            <button
              class="rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto font-bold hover:border hover:border-blue-400 focus:underline focus:underline-offset-8"
            >
              Нет
            </button>
          </div>
        </div>
        <div
          class="mx-3 mt-3 flex h-12 cursor-pointer items-center rounded border border-gray-200 bg-white shadow-md last-of-type:mb-2"
        >
          <div
            class="mx-3 cursor-pointer"
          >
            <span />
          </div>
          <p
            class="flex h-full w-full items-center overflow-x-auto italic text-gray-400 line-through"
          >
            today
          </p>
          <button
            class="mx-3 flex h-[30px] w-[30px] cursor-pointer items-center justify-center"
          >
            <span />
          </button>
        </div>
        <div
          class="mx-3 mt-3 flex h-12 cursor-pointer items-center rounded border border-gray-200 bg-white shadow-md last-of-type:mb-2"
        >
          <div
            class="mx-3 cursor-pointer"
          >
            <span />
          </div>
          <p
            class="flex h-full w-full items-center overflow-x-auto "
          >
            important
          </p>
          <button
            class="mx-3 flex h-[30px] w-[30px] cursor-pointer items-center justify-center"
          >
            <span />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overflow-y-auto [object Object],[object Object]"
    >
      <div
        class="flex h-[100px] items-center justify-center"
      >
        <form
          class="flex items-center justify-center"
        >
          <input
            class="form-input mr-3 h-10 w-52 border-none bg-transparent p-1 text-center font-roboto text-2xl placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-500"
            placeholder="Задачи"
            type="text"
            value=""
          />
          <button
            class="h-10 rounded-md bg-blue-500 px-2 py-1 font-roboto text-xl text-white shadow-md hover:bg-blue-600"
            type="submit"
          >
            Change
          </button>
        </form>
      </div>
      <form
        class="mx-3 mb-2 overflow-y-hidden rounded border border-gray-200 bg-white shadow-md h-12"
      >
        <div
          class="flex items-center py-1"
        >
          <div
            class="flex h-full w-12 items-center justify-center"
          >
            <span />
          </div>
          <div
            class="w-full"
          >
            <input
              class="form-input w-full border-none font-roboto placeholder:text-blue-500 focus:ring-transparent focus:placeholder:text-gray-300"
              name="todo"
              placeholder="Добавьте задачу"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="invisible"
        >
          <hr />
          <div
            class="flex h-10 items-center justify-end bg-gray-100"
          >
            <span />
            <span />
            <span />
            <button
              class="mx-3 rounded-md border bg-white px-2 py-[2px] font-roboto text-blue-500 hover:bg-blue-700 hover:text-white "
              type="submit"
            >
              Добавить
            </button>
          </div>
        </div>
      </form>
      <div
        class="fixed left-1/2 top-1/2 z-[2] -my-36 -ml-40 flex h-36 w-80 flex-col justify-center rounded-md border-2 border-blue-500 bg-white shadow-lg transition-all duration-200 scale-100"
      >
        <h1
          class="mb-3 text-center font-roboto text-lg font-medium"
        >
          Вы действительно хотите удалить этот список?
        </h1>
        <div
          class="flex flex-row items-center justify-around font-bold"
        >
          <a
            class="mr-2 rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto text-red-500 hover:border-blue-500 focus:underline focus:underline-offset-8"
            href="/app/"
          >
            Да
          </a>
          <button
            class="rounded-lg border border-gray-300 bg-gray-300 px-6 py-2 font-roboto font-bold hover:border hover:border-blue-400 focus:underline focus:underline-offset-8"
          >
            Нет
          </button>
        </div>
      </div>
      <div
        class="mx-3 mt-3 flex h-12 cursor-pointer items-center rounded border border-gray-200 bg-white shadow-md last-of-type:mb-2"
      >
        <div
          class="mx-3 cursor-pointer"
        >
          <span />
        </div>
        <p
          class="flex h-full w-full items-center overflow-x-auto italic text-gray-400 line-through"
        >
          today
        </p>
        <button
          class="mx-3 flex h-[30px] w-[30px] cursor-pointer items-center justify-center"
        >
          <span />
        </button>
      </div>
      <div
        class="mx-3 mt-3 flex h-12 cursor-pointer items-center rounded border border-gray-200 bg-white shadow-md last-of-type:mb-2"
      >
        <div
          class="mx-3 cursor-pointer"
        >
          <span />
        </div>
        <p
          class="flex h-full w-full items-center overflow-x-auto "
        >
          important
        </p>
        <button
          class="mx-3 flex h-[30px] w-[30px] cursor-pointer items-center justify-center"
        >
          <span />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
